src/components/LibraryContainer.tsx

```diff
@@ -112,11 +112,23 @@ export class LibraryContainer extends React.Component<LibraryContainerProps, Lib
     }
 
     refreshLibraryView(): void {
-
-        this.generatedSections = LibraryUtilities.buildLibrarySectionsFromLayoutSpecs(
+        let newSections = LibraryUtilities.buildLibrarySectionsFromLayoutSpecs(
             this.loadedTypesJson, this.layoutSpecsJson,
             this.props.defaultSectionString, this.props.miscSectionString);
 
+        // Restore item expansion
+        if (this.generatedSections) {
+            for (let previousSection of this.generatedSections) {
```

Comment:
`previousSection` should be renamed to `oldSection` (since you have `newSection`) and then also *previous* sounds like another section from a previous iteration of the loop.